tests/func/test_add.py

```diff
@@ -641,3 +641,32 @@ def test_escape_gitignore_entries(tmp_dir, scm, dvc):
 
     tmp_dir.dvc_gen(fname, "...")
     assert ignored_fname in get_gitignore_content()
+
+
+@pytest.fixture
+def subrepo(tmp_dir, scm, tmp_path_factory):
+    remote = tmp_path_factory.mktemp("subrepo_local_remote")
+
+    tmp_dir.gen({"subrepo": {}})
+    subrepo_path = tmp_dir / "subrepo"
+
+    from dvc.repo import Repo
+
+    repo = Repo.init(fspath(subrepo_path), subrepo=True)
+
+    remote_config = RemoteConfig(repo.config)
+    remote_config.add("local_remote", fspath(remote), default=True)
+
+    subrepo_path.dvc = repo
+    yield subrepo_path
+
+
+def test_parent_repo_pull(tmp_dir, scm, dvc, subrepo):
+    with subrepo.chdir():
+        subrepo.dvc_gen("file", "file content")
+        subrepo.dvc.push()
+        (subrepo / "file").unlink()
+
+    dvc.pull()
+
+    assert not (subrepo / "file").exists()
```

Comment:
This test doesn't really belong to test_add.py, right? Maybe replace it with a `Repo` test that makes sure that subrepo's dvc-files are not collected. That should be a much more local way of testing it.